src/components/common/CheckboxComponent.jsx

```diff
@@ -1,9 +1,11 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 
+import '../../_css/CheckboxComponent.css';
+
 class CheckboxComponent extends Component {
   state = {
-    isChecked: false
+    isChecked: this.props.keepCheckboxChecked(this.props.label, this.props.name) || false
```

Comment:
Having this here is a code smell. As we are moving the "checking" functionality to the store, even the state logic should be in store. In fact, `isChecked` should be passed to this component as a prop